src/packages/core/components/input-upload-field/input-upload-field.element.ts

```diff
@@ -99,11 +127,14 @@ export class UmbInputUploadFieldElement extends FormControlMixin(UmbLitElement)
 	}
 
 	render() {
-		return html`${this.#renderFiles()} ${this.#renderDropzone()}`;
+		return html`<div id="wrapper">${this.#renderFilesWithPath()} ${this.#renderFilesUploaded()}</div>
+			${this.#renderDropzone()}${this.#renderButtonRemove()}`;
 	}
 
+	//TODO When the property editor gets saved, it seems that the property editor gets the file path from the server rather than key/id.
+	// This however does not work when there is multiple files. Can the server not handle multiple files uploaded into one property editor?
 	#renderDropzone() {
-		if (!this.multiple && this._currentFiles.length) return nothing;
+		if (!this.multiple && (this._currentFiles.length || this._filePaths.length)) return nothing;
```

Comment:
Did you test with an array as you suggested when we spoke last, or perhaps the server is meant to receive one file at a time?

As we talked about, I think it is on purpose you get the URL to the file, because the file is only uploaded into the filesystem on the server thus having no unique ID, which is a feature of a database otherwise. The URL is both the unique identifier as well as the public path to the file meaning, you can load the file into an `<img />` tag for instance.